### PR TITLE
Migrate CI to Github Actions

### PR DESCRIPTION
**Pull request by deivid-rodriguez**
I was working on #494 but I noticed travis-ci.org shutdown was imminent so I figured I would migrate the CI to Github Actions.

I didn't migrate the jruby 9.0 entries, because they're being removed at #491, so I'll wait for that and then rebase this PR.

### CODE REVIEW

**Comment by olleolleolle** [comment]
@deivid-rodriguez Alright, ready!

**Comment by deivid-rodriguez** [comment]
Rebased! You'll need to enabled Github Actions in the repo settings.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  JRUBY_OPTS: -J-Xmx896M
+
+jobs:
+  ci:
+    runs-on: ubuntu-20.04
+
+    strategy:
+      matrix:
+        java-version: [8]
+        ruby-version: [jruby-9.1.17.0, jruby-9.2.17.0, jruby-head]
```
Comment (olleolleolle) [inline .github/workflows/ci.yml:21]: Ah, already a new JRuby release is out! I think this should be sufficient:

```suggestion
        ruby-version: [jruby-9.1.17.0, jruby-9.2.18.0, jruby-head]
```

**Comment by olleolleolle** [comment]
@enebo @headius Would you have access to allowing the GitHub Actions CI in Settings for this project?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (deivid-rodriguez) [reply .github/workflows/ci.yml:21]: It doesn't seem yet available on https://github.com/ruby/setup-ruby though. https://github.com/ruby/setup-ruby/pull/188 was merged but not yet released.

Comment (olleolleolle) [reply .github/workflows/ci.yml:21]: Thanks, that clarifies it. We can wait.

Comment (eregon) [reply .github/workflows/ci.yml]: It's released now

**Comment by slonopotamus** [comment]
I believe that you don't need to enable anything and GitHub Actions will start working for this repo as soon as workflow files are merged.

**Comment by deivid-rodriguez** [comment]
You need to go to "Settings > Actions", and then select an appropriate radio button under "Actions Permissions" such as "Allow all actions". Not sure whether it gets automatically enabled if workflow files land on master, but manually enabling it is better to get CI feedback beforehand and be able to merge a working setup.

**Comment by slonopotamus** [comment]
Even if you enable that (in case it is disabled which I doubt), I think GitHub Actions won't run for this PR. Because current repository doesn't contain a workflow file that would allow running builds against PRs.

**Comment by deivid-rodriguez** [comment]
I've already done this many times.

**Comment by enebo** [comment]
@olleolleolle @deivid-rodriguez I see an actions tab and it is suggesting I either pick a work flow or commit the normal GHA files to the repo.  So I think it is enabled but not displaying because we have not committed the files yet?

**Comment by deivid-rodriguez** [comment]
Yeah, that's normal. As @slonopotamus said, once we land this PR workflows should start running. Can you check the permissions enabled at "Settings > Actions"? Maybe my trick only works for branches inside the repository, not from branches in forks.

**Comment by enebo** [comment]
@deivid-rodriguez It is set to 'allow all actions' which appears to be any location and any author.

**Comment by deivid-rodriguez** [comment]
Yeah, thanks!

So indeed my trick only works for branches inside the repository (which makes sense). That's why actions are running fine on my fork without having any workflow files on master: https://github.com/deivid-rodriguez/warbler/actions. This is good enough, I'll ping you when I have something ready.

**Comment by deivid-rodriguez** [comment]
Alright, this should be ready now. Passing CI in my fork is here: https://github.com/deivid-rodriguez/warbler/actions/runs/926438002.

**Review by olleolleolle — APPROVED** [review]
Focused and clear!

**Pull request merged**